homeassistant/util/yaml.py

```diff
@@ -55,12 +55,30 @@ def _include_dir_named_yaml(loader, node):
     return mapping
 
 
+def _include_dir_merge_named_yaml(loader, node):
+    """Load multiple files from dir as a merged dict."""
+    mapping = OrderedDict()
+    files = os.path.join(os.path.dirname(loader.name), node.value, '*.yaml')
+    for fname in glob.glob(files):
+        mapping.update(load_yaml(fname))
+    return mapping
+
+
 def _include_dir_list_yaml(loader, node):
-    """Load multiple files from dir."""
+    """Load multiple files from dir as a list."""
     files = os.path.join(os.path.dirname(loader.name), node.value, '*.yaml')
     return [load_yaml(f) for f in glob.glob(files)]
 
 
+def _include_dir_merge_list_yaml(loader, node):
+    """Load multiple files from dir as a merged list."""
+    files = os.path.join(os.path.dirname(loader.name), node.value, '*.yaml')
+    merged_list = []
+    for fname in glob.glob(files):
+        merged_list += load_yaml(fname)
```

Comment:
You want to use `merged_list.extend(load_yaml(fname))` so you're not creating a new list all the time.

Please also add some validation that the loaded YAML is actually a list.